code/_onclick/xeno.dm

```diff
@@ -54,6 +54,8 @@
 					var/turf/target_turf = target
 					for(var/obj/flamer_fire/fire in target_turf)
 						firepatted = TRUE
+						if(!(src.fire_immunity & FIRE_IMMUNITY_NO_DAMAGE))
```

Comment:
```suggestion
						if(!(caste.fire_immunity & FIRE_IMMUNITY_NO_DAMAGE) || fire.tied_reagent.fire_penetrating)
```
essentially you dont need src, fireimmunity flags are in caste datums, and check if the fire is firepen regardless